### PR TITLE
Add Attachments tutorial from Daria <3

### PR DESCRIPTION
**Pull request by jhaaaa**

### CODE REVIEW

File: docs/build/add-attachments.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+## Storage considerations
+
+XMTP messages have a size limit of 1 MB and most attachment types will exceed this size. As a result, you must store attachments in a remote location outside of the XMTP network.
+
+You can use storage options such as Web3 Storage, ThirdWeb Storage, s3 buckets, and others. Another option is to allow your users to provide their own storage token.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation uses Web3 Storage, with XMTP Labs hosting with our own token. We wanted to use a decentralized solution, which made Web3 Storage a good choice. Hosting with our own token reduces user friction, but we cap uploads at 5 MB. In the future, [xmtp.chat](http://xmtp.chat) might give users the option to provide their own token if they want to upload larger file sizes or keep their images beyond a certain timeframe. 
+
+## Security considerations
+
```
Comment (jhaaaa) [inline docs/build/add-attachments.md:24]: put the following in a caution callout to emphasize the security risk

File: docs/build/add-attachments.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+:::caution
+
+Automatically loading attachments from untrusted users can have negative security implications. Be sure to consider this before enabling attachments to autoload in your app.
+
+:::
+
+A more secure pattern you might want to use is to require the user to click a “Click to load” CTA before loading an attachment. After the attachment has been loaded, it can be pulled from a cache. This pattern requires one more click for the user, but provides greater security.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation intentionally demonstrates both patterns for your reference:
```
Comment (jhaaaa) [inline docs/build/add-attachments.md:33]: tried to justify xmtp.chat supporting the less secure option as a way to intentionally show devs how to implement this.

File: docs/build/add-attachments.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+To help users avoid these error states in the first place, consider providing UI text that proactively lets users know about file size limits and supported file types.
+
+## Performance considerations
+
+To make your app more performant when loading attachments, consider caching these image URLs after the first load.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation does this using Dexie, but there are other options as well.
+
+Note that when caching attachments, you must reset the cache after the client changes. This is because the client is used to generate the attachment URL that gets added to the cache originally.
```
Comment (jhaaaa) [inline docs/build/add-attachments.md:64]: I have to admit that I don't quite understand the last sentence - but perhaps it is b/c I'm not a dev. Any thoughts? =)

File: docs/build/add-attachments.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+## Build support for message attachments
+
+As you go through the steps and code samples in this section, consider viewing the [xmtp.chat](http://xmtp.chat) reference implementation in parallel to see the features in context.
+
+### Install the package
+
```
Comment (jhaaaa) [inline docs/build/add-attachments.md:71]: Added this step - correct? =)

**Review by daria-github — APPROVED** [review]
Amazing work on this, J-Ha! 👏

File: docs/build/add-attachments.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+This document provides a step-by-step guide to providing message attachments in your app, as well as key considerations to think about when building this functionality.
+
+For a reference implementation of image attachments, see the [xmtp.chat](http://xmtp.chat) example app. 
+
+## Storage considerations
+
+XMTP messages have a size limit of 1 MB and most attachment types will exceed this size. As a result, you must store attachments in a remote location outside of the XMTP network.
+
+You can use storage options such as Web3 Storage, ThirdWeb Storage, s3 buckets, and others. Another option is to allow your users to provide their own storage token.
```
Comment (nakajima) [inline docs/build/add-attachments.md:19]: Small update:

```suggestion
You can use storage options such as Web3 Storage, ThirdWeb Storage, S3 buckets, and others. Another option is to allow your users to provide their own storage credentials for one of these services.
```

File: docs/build/add-attachments.md (hunk 2 above)
Comment (nakajima) [inline docs/build/add-attachments.md:33]:
```suggestion
The [xmtp.chat](https://xmtp.chat) reference implementation intentionally demonstrates both patterns for your reference:
```

File: docs/build/add-attachments.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+## Security considerations
+
+:::caution
+
+Automatically loading attachments from untrusted users can have negative security implications. Be sure to consider this before enabling attachments to autoload in your app.
+
+:::
+
+A more secure pattern you might want to use is to require the user to click a “Click to load” CTA before loading an attachment. After the attachment has been loaded, it can be pulled from a cache. This pattern requires one more click for the user, but provides greater security.
```
Comment (nakajima) [inline docs/build/add-attachments.md:31]: Could be good to mention the other privacy tradeoffs with automatically loading attachments, namely that the owner of the server where the attachment is stored will see user information like IP address, user agent, etc.

File: docs/build/add-attachments.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+### Create a preview attachment object
+
+Once you have the attachment object created, you can also create a preview for what to show in a message input before sending:
+
+```tsx
+URL.createObjectURL(
+              new Blob([Buffer.from(somePNGData)], {
+                type: attachment.mimeType,
+              }),
+            ),
```
Comment (nakajima) [inline docs/build/add-attachments.md:112]:
```suggestion
URL.createObjectURL(
    new Blob([Buffer.from(somePNGData)], {
    type: attachment.mimeType,
  }),
),
```

File: docs/build/add-attachments.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+### Send a remote attachment
+
+Now that you have a remote attachment, you can send it:
+
+```tsx
+await sendMessageFromHook(remoteAttachment, {
+            contentFallback: "[Attachment] Cannot display {{FILENAME}}. This app does not support attachments yet."
+            contentType: ContentTypeRemoteAttachment,
+          });
```
[the remark this answers is not in the document]
Comment (daria-github) [reply docs/build/add-attachments.md]: Good catch, this was just an example here and normally we inject this via our i18n config but wouldn't quite work in this case — 

we can change it to: 

contentFallback: `[Attachment] Cannot display ${remoteAttachment.filename}. This app does not support attachments yet.`

**Pull request merged**